week-2-wide-perspective/w02e04-dungeons-and-the-princess/src/main.ts

```diff
@@ -88,25 +25,41 @@ const knightBruce = new Person('Bruce', 'knight')
 // Próba "odbicia" królewny z podziemi lochów:
 
 // Misja ratunkowa nr 1:
-hades.initTheBarricades()
-kingdom.gotoTheDungeon(knightJohn)
-kingdom.gotoTheDungeon(knightBrienne)
-kingdom.gotoTheDungeon(kingFrancis)
+try {
+  hades.initTheBarricades()
+  kingdom.gotoTheDungeon(knightJohn)
+  kingdom.gotoTheDungeon(knightBrienne)
+  kingdom.gotoTheDungeon(kingFrancis)
+} catch (e) {
+  console.error(e)
```

Comment:
Zadanie jest wykonane w połowie. 

Dobre elementy:
- rzucanie błędów
- miejsce łapania błędów (tutaj)
- to, że otoczyłeś każdą z misji `try...catch`

Brakujące elementy:
- z tego miejsca powinieneś na jakieś podstawie określić "Skąd jest ten błąd?"
- nie chodzi tutaj o `message` wiem że w `message` jest jakaś infromacja. Ale to raczej info dla _EndUser'a_ niż dla nas.
- my chcemy np. programistycznie jakoś zadziałać jeśli błąd jest z `Castle` a inaczej jak jest z `Dungeon`

Reply:
kombinuj